uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/M2ServletContextListener.java

```diff
@@ -0,0 +1,245 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.guvnor.m2repo.backend.server;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.Enumeration;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+import javax.servlet.ServletContext;
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import javax.servlet.annotation.WebListener;
+
+import org.apache.maven.repository.internal.MavenRepositorySystemUtils;
+import org.appformer.maven.integration.Aether;
+import org.eclipse.aether.DefaultRepositorySystemSession;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.artifact.Artifact;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.connector.basic.BasicRepositoryConnectorFactory;
+import org.eclipse.aether.impl.DefaultServiceLocator;
+import org.eclipse.aether.installation.InstallRequest;
+import org.eclipse.aether.installation.InstallResult;
+import org.eclipse.aether.installation.InstallationException;
+import org.eclipse.aether.repository.LocalRepository;
+import org.eclipse.aether.spi.connector.RepositoryConnectorFactory;
+import org.eclipse.aether.spi.connector.transport.TransporterFactory;
+import org.eclipse.aether.transport.file.FileTransporterFactory;
+import org.eclipse.aether.transport.http.HttpTransporterFactory;
+import org.eclipse.aether.util.artifact.SubArtifact;
+import org.guvnor.common.services.project.model.GAV;
+import org.guvnor.m2repo.preferences.ArtifactRepositoryPreference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * It reads all the jars present in the WEB-INF/lib
+ * to create a Map with entries of GAV and path of the dependency
+ */
+@WebListener
+public class M2ServletContextListener implements ServletContextListener {
+
+    private static final Logger logger = LoggerFactory.getLogger(M2ServletContextListener.class);
+
+    private final String JAR_EXT = ".jar";
+    private final String WEB_INF_FOLDER = "WEB-INF";
+    private final String LIB_FOLDER = "lib";
+    private final String JAR_ARTIFACT = "jar";
+    private final String GROUP_ID = "groupId";
+    private final String ARTIFACT_ID = "artifactId";
+    private final String VERSION = "version";
+    private final Path tempDir;
+    private String JARS_FOLDER = File.separator + WEB_INF_FOLDER + File.separator + LIB_FOLDER + File.separator;
```

Comment:
`private static final` for constansts, please

Reply:
fixed on both branches